### ISSUE TITLE
RuntimeError: Error(s) in loading state_dict for GFPGANv1:

### ISSUE THREAD

**Issue by ntquyen11** [TencentARC/GFPGAN#65]
I train the model on my own dataset. However, when I inference image, the error happens and I don't know the reason for this
![image](https://user-images.githubusercontent.com/73950718/133723268-5d4c4721-d9fa-433d-9c65-be91e8d505e0.png)

**Comment by jcn16** [comment]
similar problem. The pretrained StyleGAN2 has size mismatch problem.

**Comment by jcn16** [comment]
you can extract pretrained StyleGAN2 from the given checkpoint, which works for me.

**Comment by ntquyen11** [comment]
Thanks for your response, I check my checkpoint and see shape that mismatches each other. However, I don't know why for that. I train my dataset with size 512x512
![image](https://user-images.githubusercontent.com/73950718/134276711-39a7cecb-a745-4ac8-9b00-1efc30c16862.png)

**Comment by ntquyen11** [comment]
> you can extract pretrained StyleGAN2 from the given checkpoint, which works for me.

Thanks for your response. I fixed this p

**Comment by xinntao** [comment]
It seems that the loaded model has a mismatched size from your configuration. 

You may need to update the configuration. 
If you need the corresponding pre-trained model, please let me know, I will upload it.

**Comment by jcn16** [comment]
To kown the reason, you can print the dict keys of  GFPGANCleanv1-NoCE-C2.pth and StyleGAN2_512_Cmul1_FFHQ_B12G4_scratch_800k.pth. The names of styleGAN mlp layer are different. 
Simply changing the keys can solve the mismatch problem

**Comment by xinntao** [comment]
> To kown the reason, you can print the dict keys of GFPGANCleanv1-NoCE-C2.pth and StyleGAN2_512_Cmul1_FFHQ_B12G4_scratch_800k.pth. The names of styleGAN mlp layer are different.
> Simply changing the keys can solve the mismatch problem

GFPGANCleanv1-NoCE-C2.pth  uses a different network from StyleGAN2_512_Cmul1_FFHQ_B12G4_scratch_800k.pth.
We have not upload the whole training process and pre-trained models for GFPGANCleanv1-NoCE-C2.pth

**Comment by ntquyen11** [comment]
> > To kown the reason, you can print the dict keys of GFPGANCleanv1-NoCE-C2.pth and StyleGAN2_512_Cmul1_FFHQ_B12G4_scratch_800k.pth. The names of styleGAN mlp layer are different.
> > Simply changing the keys can solve the mismatch problem
> 
> GFPGANCleanv1-NoCE-C2.pth uses a different network from StyleGAN2_512_Cmul1_FFHQ_B12G4_scratch_800k.pth.
> We have not upload the whole training process and pre-trained models for GFPGANCleanv1-NoCE-C2.pth

yup. I have to change channel = 1 instead of channel = 2  and arch = None when using my model. Thanks for your response
